lib/Amazon/MiqEc2InstanceTypes.rb

```diff
@@ -185,6 +185,126 @@ module MiqEc2InstanceTypes
       :vpc_only                => false,
     },
 
+    "m4.large" => {
+      :name                    => "m4.large",
+      :family                  => "General Purpose",
+      :description             => "M4 General Purpose Large",
+      :memory                  => 8.gigabytes,
+      :vcpu                    => 2,
+      :ebs_only                => true,
+      :instance_store_size     => 0,
+      :instance_store_volumes  => 0,
+      :architecture            => [:x86_64],
+      :virtualization_type     => [:hvm],
+      :network_performance     => :moderate,
+      :physical_processor      => "Intel Xeon E5-2676v3",
+      :processor_clock_speed   => 2.4, # GHz
+      :intel_aes_ni            => true,
+      :intel_avx               => true,
+      :intel_avx2              => true,
+      :intel_turbo             => true,
+      :ebs_optimized_available => true,
+      :enhanced_networking     => true,
+      :cluster_networking      => true,
+      :vpc_only                => true,
+    },
+
+    "m4.xlarge" => {
+      :name                    => "m4.xlarge",
+      :family                  => "General Purpose",
+      :description             => "M4 General Purpose Extra Large",
+      :memory                  => 16.gigabytes,
+      :vcpu                    => 4,
+      :ebs_only                => false,
```

Comment:
@Fryguy 
https://aws.amazon.com/blogs/aws/the-new-m4-instance-type-bonus-price-reduction-on-m3-c4/
Indicates that Instance Storage is EBS Only